File: sender.cs
using System;
using System.Net.Sockets;
using System.Text;

class Program
{
    const string HOST = "localhost";
    const int PORT = 65432;

    static void Main(string[] args)
    {
        Console.Write("Enter topic to publish to: ");
        string topic = Console.ReadLine();
        
        Console.Write("Enter message to publish: ");
        string message = Console.ReadLine();

        PublishToTopic(topic, message);
    }

    static void PublishToTopic(string topic, string message)
    {
        using (TcpClient client = new TcpClient(HOST, PORT))
        using (NetworkStream stream = client.GetStream())
        {
            // Create the publish message
            string publishMessage = $"publish:{topic}:{message}";
            byte[] data = Encoding.UTF8.GetBytes(publishMessage);

            // Send the publish message to the broker
            stream.Write(data, 0, data.Length);
            Console.WriteLine($"Sent to topic \"{topic}\": {message}");
        }
    }
}

File: sender.py
import socket

HOST = 'localhost'
PORT = 65432


def publish_to_topic(topic, message):
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client_socket:
        # Connect to the broker
        client_socket.connect((HOST, PORT))

        # Send a publish request with the topic and message
        publish_message = f'publish:{topic}:{message}'
        client_socket.sendall(publish_message.encode())
        print(f'Sent to topic "{topic}": {message}')


if __name__ == "__main__":
    topic = input("Enter topic to publish to: ")
    message = input("Enter message to publish: ")
    publish_to_topic(topic, message)
